pkg/routeagent-driver/handlers/kubeproxy-iptables/kp-iptables.go

```diff
@@ -0,0 +1,67 @@
+package kp_iptables
+
+import (
+	"fmt"
+	"os"
+
+	"k8s.io/klog"
+
+	"github.com/submariner-io/submariner/pkg/event"
+	cni_driver "github.com/submariner-io/submariner/pkg/routeagent-driver/cni-driver"
+)
+
+type SyncHandler struct {
+	event.HandlerBase
+	clusterID        string
+	namespace        string
+	localClusterCidr []string
+	localServiceCidr []string
+
+	hostname string
+	cniIface *cni_driver.CniInterface
+}
+
+func NewSyncHandler(clusterID, namespace string, clusterCidr, serviceCidr []string) *SyncHandler {
+	return &SyncHandler{
+		clusterID:        clusterID,
+		namespace:        namespace,
+		localClusterCidr: clusterCidr,
+		localServiceCidr: serviceCidr,
+	}
+}
+
+func (kp *SyncHandler) GetName() string {
+	return "kubeproxy-iptables-handler"
+}
+
+func (kp *SyncHandler) GetNetworkPlugin() string {
+	return event.AnyNetworkPlugin
+}
+
+func (kp *SyncHandler) Init() error {
+	var err error
+	kp.hostname, err = os.Hostname()
+	if err != nil {
+		klog.Fatalf("unable to determine hostname: %v", err)
+	}
+
+	cniIface, err := cni_driver.DiscoverCNIInterface(kp.localClusterCidr[0])
```

Comment:
I would move this to the main of the route agent, since we will most likely leed the CNI interface details always for any driver. Other services rely on this tag being present on the node. Even the test? I can't remember what exactly but it's something we expect.

Reply:
As I explained above, AFAIU this is specific to kp-iptables driver. I'm not sure we require this for OVN - please let me know.

I'll add similar code in the main() method of route-agent where we discover and annotate the respective node - this piece of code seems to be common to all CNIs and can be re-used.

Reply:
Makes sense Sridhar